modules/terrain/src/lib/parse-terrain.js

```diff
@@ -50,34 +50,36 @@ function getMeshAttributes(vertices, terrain, tileSize, bounds) {
     positions[3 * i + 1] = -y * yScale + maxY;
     positions[3 * i + 2] = terrain[pixelIdx];
 
-    texCoords[2 * i + 0] = x / tileSize;
-    texCoords[2 * i + 1] = y / tileSize;
+    texCoords[2 * i + 0] = x / width;
+    texCoords[2 * i + 1] = y / height;
   }
 
   return {
-    POSITION: {value: positions, size: 3},
-    TEXCOORD_0: {value: texCoords, size: 2}
+    POSITION: { value: positions, size: 3 },
+    TEXCOORD_0: { value: texCoords, size: 2 }
     // NORMAL: {}, - optional, but creates the high poly look with lighting
   };
 }
 
-function getMartiniTileMesh(terrainImage, terrainOptions) {
+function getMesh(terrainImage, terrainOptions) {
   if (terrainImage === null) {
     return null;
   }
-  const {meshMaxError, bounds, elevationDecoder} = terrainOptions;
+  const { meshMaxError, bounds, elevationDecoder } = terrainOptions;
 
   const data = terrainImage.data;
-  const tileSize = terrainImage.width;
-  const gridSize = tileSize + 1;
+  const width = terrainImage.width;
+  const height = terrainImage.height;
+  const terrain = getTerrain(data, width, height, elevationDecoder);
 
-  const terrain = getTerrain(data, tileSize, elevationDecoder);
-
-  const martini = new Martini(gridSize);
-  const tile = martini.createTile(terrain);
-  const {vertices, triangles} = tile.getMesh(meshMaxError);
-
-  const attributes = getMeshAttributes(vertices, terrain, tileSize, bounds);
+  let mesh;
+  if (terrainOptions.tesselector === 'martini' && width === height) {
```

Comment:
Should we call the option `tesselator` instead of `tesselector`? This would be consistent with usage in e.g. deck.gl, and I have never seen tesselector being used.

IMHO, it is a bit weird to have this auto fallback silently switching to `delatin`. An alternative approach that is consistent with other loaders would be to support three options: `auto`, `martini` and `delatin`, list them in a simple switch statement, and throw an error if `martini` is specified and does not work for the input data.

`auto` could be the default if not specified.

Reply:
Also note you still are just checking if the image is a square, and you also need to check if it's a power of 2. https://github.com/visgl/loaders.gl/pull/1372#discussion_r629829407

Reply:
got it, let me add that

Reply:
one question, auto means using martini right?